src/transformers/modeling_utils.py

```diff
@@ -1718,6 +1718,12 @@ def save_pretrained(
             for ignore_key in self._keys_to_ignore_on_save:
                 if ignore_key in state_dict.keys():
                     del state_dict[ignore_key]
+        # Disable to see the damage.
+        if safe_serialization:
+            if self._keys_to_ignore_on_load_missing is not None:
+                for ignore_key in self._keys_to_ignore_on_load_missing:
+                    if ignore_key in state_dict.keys():
+                        del state_dict[ignore_key]
```

Comment:
This is the core of the fix:

I think of 3 ways to get this done:

- Just add `_keys_to_ignore_on_save` for all affected models. No change in core modeling code. Will impact both torch and safetensors saving versions (so technically breaking change for the on-disk representation, should users reuse the dictionnairies in non-transformers modeling code). Essentially will have to maintain both `_keys_to_ignore_on_save` AND `_keys_to_ignore_on_load_missing` at the same time for all models

- [Proposed fix] Ignore the `ignore_on_load` **only** for safetensors. The benefit is that it's not breaking for torch, and still allows saving with safetensors for all models. However it does introduce a difference in mecanism for both on disk representation

- Ignore the `ignore_on_load` for **both** torch and safetensors. Will have the same the same effectr as the first proposed fix, with the benefit of nto having to maintain both keys. Technically both are not exactly the same because sometimes models have been renamed leading to old names being in those keys while not being used anymore. This is more true for the third key `_keys_to_ignore_extra` but still could potentially exist for `ignore_on_save` and `ignore_on_load`. I think this is ok, because the current code will just ignore keys not contained in the state dict.

I propose the second change since it's the least breaking now, even though I think in the long run it could cause more issues/confusion because of the dichotomy in treatment.

In any case, all proposed changes would only affect *future* models being created, and have no effect on currently existing models in the wild.